### ISSUE TITLE
Empty pull request list item in issue 422

### ISSUE THREAD

**Issue by kraai** [rust-lang/this-week-in-rust#2781]
The tenth pull request list item in [issue 422](https://this-week-in-rust.org/blog/2021/12/22/this-week-in-rust-422/) is empty.  The original text is

```
* [add `[T]::as_simd(_mut)`](https://github.com/rust-lang/rust/pull/91479)
```

I think that the Markdown formatter interprets the item body as a reference link definition for <code>add `T</code>.

**Comment by ericseppanen** [comment]
This formatting glitch is really weird. I tried to figure out what's happening:
- It's the right-bracket `]` that is the problem
- but only when it's followed by a colon `]:`

I can't find any explanation of why this doesn't work; the python `commonmark` module handles this correctly, so I can only conclude that it's a bug in the `Markdown` library.

**Comment by ericseppanen** [comment]
As a really dirty hack, if you put a unicode zero-width space character in between the `]` and `:`, then it no longer triggers the bug.

**Comment by ericseppanen** [comment]
Here's a slightly less hacky way to work around the Markdown bug: just add something else to the list item. For example, adding an html space `&nbsp;` at the end of the affected line will stop the bug from happening.

**Comment by ericseppanen** [comment]
Fixed by #2784.
